### DISCUSSION TITLE
Simple convert of dict-constructor to dict-literal, comply with PyLint use-dict-literal / R1735

### DISCUSSION THREAD

**Discussion by dmakhno** [microsoft/pylance-release#6227] [Enhancement]
Due to performance reasons pylint suggest a lot to refactor dict to literal form.
https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/use-dict-literal.html

It is straight forward action in most case, but human-time consuming. 
Would be nice to have it from the box. 

I cannot find any extension that supports it, neither other FRs.
Also I used other "brainy" IDE, where this feature exist for a long time, and very missing it vscode. At least in form to make pylint happy.

**Comment by dmakhno** [comment]
Additional repro to mentioned in pylint link...

```python
a = dict(
    b=dict(
        c=1,
        d=2,
    ),
    l=[
        dict(
            x=3,
            y=4,
        )
    ]

)
```
complains
```
R1735: Consider using '{"b": dict(c=1, d=2), "l": [dict(x=3, y=4)]}' instead of a call to 'dict'. (use-dict-literal)
R1735: Consider using '{"c": 1, "d": 2}' instead of a call to 'dict'. (use-dict-literal)
R1735: Consider using '{"x": 3, "y": 4}' instead of a call to 'dict'. (use-dict-literal)
```

Expectations of pylint, and this FR to be able it from 
```python
a = {
   "b": ...,
   "l": [...],
}

# b as
{
  "c": 1,
  "d": 2,
}

# l item as 
{
  "x": 3,
  "y": 4,
}
```

It can be recursive - once for root
or three user actions - once per dict

**Comment by rchiodo** [comment]
Transferring to discussions for up votes.
